### PR TITLE
Define module we can use to safely rescue most exceptions.

### PR DESCRIPTION
**Pull request by myronmarston**

### CODE REVIEW

**Comment by myronmarston** [comment]
Anyone want to review this?  It supports rspec/rspec-expectations#844 and rspec/rspec-core#2063.

**Comment by myronmarston** [comment]
I'd love to get this merged before leaving on my 3-week vacation tomorrow.  Anyone want to review this, @rspec/rspec ?

Bear in mind that the failures from the rspec-core and rspec-expectations PRs are due to a bug in our build scripts -- see rspec/rspec-dev#137.

**Comment by xaviershay** [comment]
looking...

File: lib/rspec/support.rb [hunk 1]
```diff
@@ -118,6 +118,17 @@ def self.warning_notifier
       @warning_notifier ||= DEFAULT_WARNING_NOTIFIER
     end
 
+    # @private
+    module AllExceptionsExceptOnesWeMustNotRescue
+      # These exceptions are dangerous to rescue as rescuing them
+      # would interfere with things we should not interfere with.
```
Comment (xaviershay) [inline lib/rspec/support.rb]: lol, great name and comment

File: spec/rspec/support_spec.rb [hunk 2]
```diff
@@ -207,5 +207,42 @@ def append_to_array_instead_of_warning
...
+      it 'rescues an Exception' do
+        expect {
+          begin
+            raise Exception
+          rescue subject
+          end
+        }.not_to raise_error
+      end
+
+      Support::AllExceptionsExceptOnesWeMustNotRescue::AVOID_RESCUING.each do |klass|
```
Comment (xaviershay) [inline spec/rspec/support_spec.rb]: not a huge fan of this but spelling them all out probably gross too

**Comment by xaviershay** [comment]
LGTM

**Comment by myronmarston** [comment]
Thanks, @xaviershay.

**Pull request merged**